### DISCUSSION TITLE
Can someone explain `JSONParsed` and why my object becomes a string?

### DISCUSSION THREAD

**Discussion by mmlngl** [honojs/hono#1824] [Q&A]
I've really enjoyed playing with Honojs.
I built an API that returns `NoteModel[]`.

the return type is interfered as:
```
JSONParsed<{
    ok: boolean;
    notes: NoteModel[];
}
```

When I list the contents in my react app, I see the `NoteModel` data, but typescript infers it as a string. I'm using react-query for reference.

This is clearly by design, I just looking to understand why.

Then a follow-up: how am I supposed to loop of this data and use it in my frontend app?

**Comment by jnatmorris** [comment]
I had the same problem. I found that for myself it was due to the nesting of interfaces. But then when messily combining all of my nested interfaces into a single interface it worked.

For your example it would be:
`interface obj {
   ok: boolean;
   notes: NoteModel[];
}
`
`interface NoteModel{
   a: ...
   b: ...
}
`
converted into: 
`
interface obj {
   ok: boolean;
   notes: {
       a: ...
       b: ...
   }[];
}
`

**Reply by jnatmorris** [reply]
Forgot to say, I believe this is a bug rather than a feature.

**Comment by yusukebe** [comment]
To change this, we need to modify the `JSONParsed` type. I may work on it, but I don't have time, so if you can do it, please PR. However, please make sure that the `Date` type, etc. becomes a `string` when `JSON.stringify` is done.
